user/jwtValidator.go

```diff
@@ -0,0 +1,152 @@
+package user
+
+import (
+	"errors"
+	"log"
+	"net/http"
+	"strings"
+
+	auth0 "github.com/auth0-community/go-auth0"
+	jwt "github.com/dgrijalva/jwt-go"
+	jose "gopkg.in/square/go-jose.v2"
+)
+
+type JWTValidator interface {
+	ValidateRequest(request *http.Request) (*TokenData, error)
+}
+
+type jwtValidator struct {
+	config JWTValidatorConfig
+}
+
+type JWTValidatorConfig struct {
+	Auth0Domain string
+	Secret      string
+}
+
+func NewJWTValidator(config JWTValidatorConfig) (*jwtValidator, error) {
```

Comment:
Don't change this now, but for only two fields, I wouldn't bother with the separate config struct. Just pass them both directly.